spec/01-lexical-syntax.md

```diff
@@ -82,8 +82,10 @@ decomposes into the three identifiers `big_bob`, `++=`, and
 The rules for pattern matching further distinguish between
 _variable identifiers_, which start with a lower case letter, and
 _constant identifiers_, which do not. For this purpose,
-underscore `‘_‘` is taken as lower case, and the ‘\$’ character
-is taken as upper case.
+all letters in unicode category Lo (lowercase letter), and all letters that 
+that have contributory property Other_Lowercase that are not in
+category Nl (letter numerals) are taken as lower case, as well as
+underscore `‘_‘`.
```

Comment:
Unicode is a proper noun and starts with a Lu.

I'm not an expert, so this rephrasing is not very clarifying to me. To me, who kind of knows what lower case means, the news is that underscore is and dollar isn't. (But for all I know, taking underscore was an implementation error.)

Assuming the info is correct, I'd suggest the phrasing: "More formally, variable identifiers may start with any of the following: all letters...; all letters that have...; and underscore. Note that this category excludes `$`."

Don't copy/paste the double "that". I might edit the third that to "and which", or at least "and that".

If this is important enough to spec, I would add illustrative examples. I would also add a test. Since I don't really know what you're talking about, I don't if it's true or will still hold in dotty or anything. I checked some modifier letters but some worked and some didn't, apparently they come in flavors.

Reply:
Good points.

I still expect someone to show up and argue the implementation should be changed, and a third person to disagree how the implementation should be changed, but examples and tests that describe the status quo would help rather than hinder that. I'll get on that.

Reply:
I moved the `_` out of the lower case category, and added is as an exception to the variable identifier syntax. Then I can just talk about the categories that contain characters that have the Other_Lowercase property.

I didn't think "more formally" was the phrasing to go with -- it's more precise, but not really more formal. A "Permission to zone out" phrasing wasn't really required at all IMO, this is the spec after all. If it's important enough to implement, it ought to be important enough to spec.